packages/next/src/server/app-render/app-render.tsx

```diff
@@ -218,13 +220,19 @@ function parseRequestHeaders(
   headers: IncomingHttpHeaders,
   options: ParseRequestHeadersOptions
 ): ParsedRequestHeaders {
+  const isDevWarmupRequest = headers[NEXT_DEV_WARMUP_HEADER] !== undefined
+
+  // dev warmup requests are treated as prefetch RSC requests
   const isPrefetchRequest =
+    isDevWarmupRequest ||
     headers[NEXT_ROUTER_PREFETCH_HEADER.toLowerCase()] !== undefined
 
   const isHmrRefresh =
     headers[NEXT_HMR_REFRESH_HEADER.toLowerCase()] !== undefined
 
-  const isRSCRequest = headers[RSC_HEADER.toLowerCase()] !== undefined
+  // dev warmup requests are treated as prefetch RSC requests
```

Comment:
```suggestion
  // dev warmup requests are treated as RSC requests
```

Reply:
I mean to convey that warmups are both prefetch and RSC (and we're setting RSC here and prefetch in the other line)

Reply:
Yeah my nitpick was just that the comment is saying warmups are prefetch RSC requests above the non prefetch case.